File: src/main/java/gargoyle/ct/pref/CTPropertyChangeEvent.java
package gargoyle.ct.pref;

import gargoyle.ct.prop.CTProperty;
import org.jetbrains.annotations.NotNull;

import java.text.MessageFormat;

public class CTPropertyChangeEvent<T> {
    private final String name;
    private final T newValue;
    private final T oldValue;
    private final CTProperty<T> property;

    public CTPropertyChangeEvent(CTProperty<T> property, String name, T oldValue, T newValue) {
        this.property = property;
        this.name = name;
        this.oldValue = oldValue;
        this.newValue = newValue;
    }

    public String getName() {
        return name;
    }

    public T getNewValue() {
        return newValue;
    }

    public T getOldValue() {
        return oldValue;
    }

    public CTProperty<T> getProperty() {
        return property;
    }

    @NotNull
    @Override
    public String toString() {
        return MessageFormat.format("CTPropertyChangeEvent'{'property={0}'}'", property);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/CTPropertyChangeEvent.kt
package gargoyle.ct.pref

import gargoyle.ct.prop.CTProperty
import java.text.MessageFormat

class CTPropertyChangeEvent<T : Any>(val property: CTProperty<T>, val name: String, val oldValue: T, val newValue: T) {

    override fun toString(): String {
        return MessageFormat.format("CTPropertyChangeEvent'{'property={0}'}'", property)
    }
}
